src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,464 @@
+                "lengthy collection of metrics. " +
+                "" +
+                "The tool takes in a VCF for the sample " +
+                "to skip actually polymorphic bases. As a result it assumes that any disagreement from the reference is an error." +
+                "" +
+                "This assumption can, of-course be wrong, but the rates at which it is wrong is likely to be lower than" +
+                "the error rates that are exposed. " +
+                "" +
+                "The program also takes a region over which to run. This region should be used to define a 'confidence region' " +
+                "where lack of a call in the VCF translates to a high probability of a monomorphic site.",
+
+        oneLineSummary = "Program to collect error metrics on bases stratified in various ways.",
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+public class CollectBamErrorMetrics extends CommandLineProgram {
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "Input SAM or BAM file.")
+    public File INPUT;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "Basename for output files.")
+    public File OUTPUT;
+
+    @Argument(doc = "Errors to collect in the form of \"ERROR(,STRATIFIER)*\". " +
+            "To see the values available for ERROR and STRATIFIER look at the documentation for the arguments " +
+            "ERROR_VALUE and STRATIFIER_VALUE.")
+    public List<String> ERROR_METRICS = new ArrayList<>(CollectionUtil.makeList(
+            "ERROR",
+            "ERROR,BASE_QUALITY",
+            "ERROR,INSERT_LENGTH",
+            "ERROR,GC_CONTENT",
+            "ERROR,READ_DIRECTION",
+            "ERROR,PAIR_ORIENTATION",
+            "ERROR,HOMOPOLYMER",
+            "ERROR,BINNED_HOMOPOLYMER",
+            "ERROR,CYCLE",
+            "ERROR,READ_ORDINALITY",
+            "ERROR,READ_ORDINALITY,CYCLE",
+            "ERROR,READ_ORDINALITY,HOMOPOLYMER",
+            "ERROR,READ_ORDINALITY,GC_CONTENT",
+            "ERROR,READ_ORDINALITY,PRE_DINUC",
+            "ERROR,MAPPING_QUALITY",
+            "ERROR,READ_GROUP",
+            "ERROR,MISMATCHES_IN_READ",
+            "ERROR,ONE_BASE_PADDED_CONTEXT",
+            "OVERLAPPING_ERROR",
+            "OVERLAPPING_ERROR,BASE_QUALITY",
+            "OVERLAPPING_ERROR,INSERT_LENGTH",
+            "OVERLAPPING_ERROR,READ_ORDINALITY",
+            "OVERLAPPING_ERROR,READ_ORDINALITY,CYCLE",
+            "OVERLAPPING_ERROR,READ_ORDINALITY,HOMOPOLYMER",
+            "OVERLAPPING_ERROR,READ_ORDINALITY,GC_CONTENT"));
+
+    @Argument(doc = "A fake argument used to show what the options of ERROR (in ERROR_METRICS) are.", optional = true)
+    public BaseErrorCalculation.Errors ERRORS_VALUE = null;
+
+    @Argument(doc = "A fake argument used to show what the options of STRATIFIER (in ERROR_METRICS) are.", optional = true)
+    public ReadBaseStratification.Stratifiers STRATIFIER_VALUE = null;
```

Comment:
I believe that there is an easier way to handle this with barclay - maybe a plugin ([`CommandLinePluginDescriptor`](https://github.com/broadinstitute/barclay/blob/master/src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java)) or a tagged-argument ([`TaggedArgument`](https://github.com/broadinstitute/barclay/blob/master/src/main/java/org/broadinstitute/barclay/argparser/TaggedArgument.java))? Or a combination of both?